### PR TITLE
Option to split root TOC into multiple child TOC files

### PR DESCRIPTION
**Pull request by rwrife**
Added an option to split the root TOC into multiple child TOC files for each child folder down to a certain depth in the directory structure.  Where adding -m 0 indicates that there should be no child TOC files created and only one large TOC at the root (the default) should be generated, -m 1 indicates to go one-level deep in the directory traversal, creating a toc at the root level that has hrefs that point the child folders, which themselves contain a TOC with remaining child structure defined.  This allows for small TOC file at the root with easier management of child TOCs on large projects.

### CODE REVIEW

**Comment by asishskaria** [comment]
it is an awesome feature I am waiting so that I onboard huge docfx projects with humongous toc files without killing my ux.

**Review by Ellerbach — APPROVED** [review]
This looks good to me. I'm curious to see how this will result.
And can you also update the README with this new option?

**Comment by mtirionMSFT** [comment]
Agree with @Ellerbach , please also modify the README with this new option.

**Comment by rwrife** [comment]
I updated the readme.

File: src/DocFxTocGenerator/README.md [hunk 1]
```diff
@@ -7,15 +7,16 @@ This tool allow to generate a yaml compatible `toc.yml` file for DocFX.
...
--d, --docfolder       Required. Folder containing the documents.
--o, --outputfolder    Folder to write the resulting toc.yml in.
--v, --verbose         Show verbose messages.
--s, --sequence        Use the .order files for TOC sequence. Format are raws of: filename-without-extension
--r, --override        Use the .override files for TOC file name override. Format are raws of: filename-without-extension;Title you want
--i, --index           Auto-generate a file index in each folder.
--g, --ignore          Use the .ignore files for TOC directory ignore. Format are raws of directory names: directory-to-ignore
---help                Display this help screen.
---version             Display version information.
+-d, --docfolder			Required. Folder containing the documents.
```
Comment (Ellerbach) [inline src/DocFxTocGenerator/README.md:10]: Can you please properly align the text like it was done with the previous text?
Use spaces and not tabs (not sure what's use).
image

**Review by Ellerbach — COMMENTED** [review]
On little detail and good to go!

File: src/DocFxTocGenerator/README.md (hunk 1 above)
Comment (rwrife) [reply src/DocFxTocGenerator/README.md:10]: Looks like this got merged already, but I'll fix the readme, I had to push the descriptions over to make room for the <depth> parameter...not sure if that's the proper way of documenting param inputs or not

**Pull request merged**